src/custom/components/AffiliateStatusCheck/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import { useActiveWeb3React } from 'hooks/web3'
+import NotificationBanner from 'components/NotificationBanner'
+import { useReferralAddress, useResetReferralAddress } from 'state/affiliate/hooks'
+import { updateAppDataHash } from 'state/affiliate/actions'
+import { useAppDispatch } from 'state/hooks'
+import { hasTrades } from 'utils/trade'
+import { uploadAppDataDoc } from 'utils/operator'
+import { generateReferralMetadataDoc } from 'utils/metadata'
+
+type AffiliateStatus = 'NOT_CONNECTED' | 'OWN_LINK' | 'ALREADY_TRADED' | 'ACTIVE'
+
+const STATUS_TO_MESSAGE_MAPPING: Record<AffiliateStatus, string> = {
+  NOT_CONNECTED: 'Please connect your wallet to participate',
+  OWN_LINK: 'You followed your own referral link',
+  ALREADY_TRADED: 'You have already traded with the current account',
+  ACTIVE: 'Your affiliate link will be effective on next trade',
+}
+
+export default function AffiliateStatusCheck() {
+  const appDispatch = useAppDispatch()
+  const resetReferralAddress = useResetReferralAddress()
+  const history = useHistory()
+  const { account, chainId } = useActiveWeb3React()
+  const referralAddress = useReferralAddress()
```

Comment:
Yep, i understand that part.

I'm pro persisting it also. See my proposal for the `AffiliateState` above. My point is that `appDataHash` and `referralAddress` would be just set only at the same time, after all validations are done. 
They are part of the same referal object, because they are related (either both are set, or none).

All the other things, afait, they can be handed by getting the address from the URL and removing it when necessary.

Reply:
I see it slightly different.
`referralAddress` is set in a prior step, so you can have `referralAddress` without ending up having `appDataHash` ever set.
Think of the use case of a user following own address for instance.
The user can still use the same address with another account if they wish to.